sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,417 @@
+                        return
+                    except Exception as e:
+                        self._log.error("While executing fetch schemas for databse - %s, the following exception occured - %s", db_name, e)
+                # Switch DB back to MASTER
+                if not is_azure_sql_database(engine_edition):
+                    cursor.execute(SWITCH_DB_STATEMENT.format(self._check.connection.DEFAULT_DATABASE))
+   
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _collect_schemas_data(self):
+        """Collects database information and schemas and submits to the agent's queue as dictionaries
+        schema dict
+        key/value:
+            "name": str
+            "id": str
+            "owner_name": str
+            "tables" : list of tables dicts
+                table
+                key/value:
+                    "id" : str
+                    "name" : str
+                    columns: list of columns dicts
+                        columns
+                        key/value:
+                            "name": str
+                            "data_type": str
+                            "default": str
+                            "nullable": bool
+                indexes : list of index dicts
+                    index
+                    key/value:
+                        "name": str
+                        "type": str
+                        "is_unique": bool
+                        "is_primary_key": bool
+                        "is_unique_constraint": bool
+                        "is_disabled": bool,
+                        "column_names": str
+                foreign_keys : list of foreign key dicts
+                    foreign_key
+                    key/value:
+                        "foreign_key_name": str
+                        "referencing_table": str
+                        "referencing_column": str
+                        "referenced_table": str
+                        "referenced_column": str
+                partitions: partition dict
+                    partition
+                    key/value:
+                        "partition_count": int
+        """
+        self._data_submitter.reset()
+        self._data_submitter.set_base_event_data(
+            self._check.resolved_hostname,
+            self._check.non_internal_tags,
+            self._check._config.cloud_metadata,
+            "{},{}".format(
+                self._check.static_info_cache.get(STATIC_INFO_VERSION, ""),
+                self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, ""),
+            ),
+        )
+
+        databases = self._check.get_databases()
+        db_infos = self._query_db_informations(databases)
+        self._data_submitter.store_db_infos(db_infos)
+        self._fetch_for_databases()
+        self._log.debug("Finished collect_schemas_data")
```

Comment:
nit: should this log line be _after_ submit executes?

Reply:
thx, done